Hoje eu aprendi sobre Exceptions e Try Catch.

EN: Today I've Learned about Exceptions / Try Catch

Code:
void main() {
  try{
    int resultado = 100 ~/ 2;
    print(resultado);
    
    double? valor = double.tryParse('50.2a');
    print(valor);
  } on UnsupportedError {
    print('Erro unsuported');
  } on FormatException catch (error) {
    print('Erro Format ${error.message}');
  } catch (e) {
    print (e);
  } finally {
    print('finalizou');
  }
}
